src/app/features/psbt-signer/hooks/use-parsed-psbt.tsx

```diff
@@ -0,0 +1,103 @@
+import { useCallback, useMemo } from 'react';
+
+import * as btc from '@scure/btc-signer';
+
+import { isDefined } from '@shared/utils';
+
+import { subtractMoney } from '@app/common/money/calculate-money';
+import { useCurrentAccountNativeSegwitIndexZeroSigner } from '@app/store/accounts/blockchain/bitcoin/native-segwit-account.hooks';
+import { useCurrentAccountTaprootIndexZeroSigner } from '@app/store/accounts/blockchain/bitcoin/taproot-account.hooks';
+import { useCurrentNetwork } from '@app/store/networks/networks.selectors';
+
+import { useInputsOutputsTotals } from './use-inputs-outputs-totals';
+import { useMatchedSats } from './use-matched-sats';
+import { useParsedInputs } from './use-parsed-inputs';
+import { useParsedOutputs } from './use-parsed-outputs';
+
+interface UseParsedPsbtArgs {
+  allowedSighashes?: btc.SignatureHash[];
+  inputs: btc.TransactionInput[];
+  inputsToSign?: number | number[];
+  outputs: btc.TransactionOutput[];
+}
+export function useParsedPsbt({
+  allowedSighashes,
+  inputs,
+  inputsToSign,
+  outputs,
+}: UseParsedPsbtArgs) {
+  const network = useCurrentNetwork();
+  const bitcoinAddressNativeSegwit = useCurrentAccountNativeSegwitIndexZeroSigner().address;
+  const { address: bitcoinAddressTaproot } = useCurrentAccountTaprootIndexZeroSigner();
+  const { isPsbtMutable, parsedInputs } = useParsedInputs({
+    allowedSighashes,
+    inputs,
+    inputsToSign,
+  });
+  const parsedOutputs = useParsedOutputs({ isPsbtMutable, outputs, network });
+  const matchedSatsWithOutputs = useMatchedSats(parsedInputs, parsedOutputs);
+
+  const {
+    addressNativeSegwitInputTotal,
+    addressTaprootInputTotal,
+    addressNativeSegwitOutputTotal,
+    addressTaprootOutputTotal,
+  } = useInputsOutputsTotals({
+    bitcoinAddressNativeSegwit,
+    bitcoinAddressTaproot,
+    parsedInputs,
+    parsedOutputs,
+  });
+
+  const matchedSatsWithInscriptions = useMemo(() => {
+    return matchedSatsWithOutputs.matchedSats.filter(match =>
+      match.inputs.find(utxo => {
+        if (isDefined(utxo.inscription)) return utxo.inscription;
```

Comment:
cannot it be just 
```
return utxo.inscription
```

Reply:
Yep, haha, thanks. Not sure how I ended up with that one. 🙃